File: prova/backend/src/main/java/com/pw/cadastro/services/AnimalService.java
package com.pw.cadastro.services;

import com.pw.cadastro.interfaces.IAnimalService;
import com.pw.cadastro.models.Animal;
import com.pw.cadastro.repository.AnimalRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.CrossOrigin;

import java.util.List;

@Service
@CrossOrigin(origins = "*")
public class AnimalService implements IAnimalService {

    @Autowired
    AnimalRepository animalRepository;

    @Override
    public Animal cadastrar(Animal animal) {
        return animalRepository.save(animal);
    }

    @Override
    public List<Animal> listar() {
        return animalRepository.findAll();
    }
}

File: prova/frontend/src/services/animalService.js
import { ApiService } from './api.service'

const animalService = {

    listar: async() => {
        return ApiService.get('animal/listar');
    },

    cadastrar: async(animal) => {
        return ApiService.post('animal/cadastrar', animal);
    }
}

export default animalService;